### PR TITLE
Improve handling of unknown or unexpected bugrefs

### PR DESCRIPTION
**Pull request by perlpunk**
e.g. `jsc#ABC-123`

Issue: https://progress.opensuse.org/issues/95989

Additionally improve logging in case the regex does not match, so that
we immediately see what didn't match and why.

If necessary, I will try to add a test, but will be afk for a few hours and wanted to push a fix ASAP.

### CODE REVIEW

File: openqa_review/openqa_review.py [hunk 1]
```diff
@@ -282,8 +282,14 @@ def get_test_bugref(entry):
     bugref = entry.find(id=re.compile("^bug-"))
     if not bugref:
         return {}
+    reference = re.search(r"\S+#((?:[A-Z]+-)?[0-9]+)", bugref.i["title"])
+    if reference:
+        reference = reference.group()
+    else:
+        sys.stderr.write("Error: Could not find bugref in '%s'\n" % bugref.i["title"])
+        reference = "???"
```
Comment (okurz) [inline openqa_review/openqa_review.py:290]: We have a logging object which you should be able to use.
Also tests failed as this section is uncovered: https://app.travis-ci.com/github/os-autoinst/openqa_review/jobs/526835360#L252
As you handle the error already and provide the necessary results in the output I suggest to not treat it as an "error" from perspective of openqa-review as the mitigation is already automatically applied so we should rather make the reference text explicit. How about:

```suggestion
    reference = re.search(r"\S+#((?:[A-Z]+-)?[0-9]+)", bugref.i["title"])
    bugref = reference.group() if reference else "(unknown bugref in %s)" % bugref.i["title"]"
```

As an alternative to the above we could mark the section explicitly as acceptable to not be covered but I would very much prefer the above

**Review by okurz — CHANGES_REQUESTED** [review]
I would prefer to only introduce error handling for unknown bugrefs for now. We should not pretend to support Jira tickets when we can not read out their status in another step.

**Comment by perlpunk** [comment]
> I would prefer to only introduce error handling for unknown bugrefs for now. We should not pretend to support Jira tickets when we can not read out their status in another step.

Can you rephrase that? I don't understand what you mean here.

**Comment by okurz** [comment]
> I would prefer to only introduce error handling for unknown bugrefs for now. We should not pretend to support Jira tickets when we can not read out their status in another step.

Rephrasing: I suggest you only introduce the handling of unexpected bugref formats including the suggestion I made in https://github.com/os-autoinst/openqa_review/pull/173#discussion_r676465447 . I suggest to not change the regex to cover Jira tickets as in other place where we read out information from tickets we do not support Jira, e.g. when reading the status, priority, assignee.

**Review by okurz — APPROVED** [review]
